src/main/java/gregtech/api/recipes/RecipeBuilder.java

```diff
@@ -95,9 +102,27 @@ protected RecipeBuilder(RecipeBuilder<R> recipeBuilder) {
         this.EUt = recipeBuilder.EUt;
         this.hidden = recipeBuilder.hidden;
         this.onBuildAction = recipeBuilder.onBuildAction;
+        this.cleanroom = recipeBuilder.cleanroom;
+    }
+
+    public R cleanroom(@Nullable CleanroomType cleanroom) {
```

Comment:
If this is how you are doing this, `CleanroomType` needs to be exposed to Crafttweaker, or you need to have a method in CTRecipeBuilder that takes string `CleanroomType` keys and then passes the `CleanroomType` object to the backingBuilder.